azure-defender-easm/scripts/kusto_queries.py

```diff
@@ -0,0 +1,108 @@
+from kusto_client import KUSTO_CLIENT, KUSTO_DATABASE
+from azure.kusto.data.helpers import dataframe_from_result_table
+
+
+def get_host_asset(host_name):
+    query = f"EasmHostAsset | where Host == '{host_name}' | limit 1"
+    response = KUSTO_CLIENT.execute(KUSTO_DATABASE, query)
+    data = dataframe_from_result_table(response.primary_results[0]).to_dict(
+        orient="records"
+    )
+    return data
+
+
+def get_page_asset(page_name):
+    query = f"EasmPageAsset | where Host == '{page_name}' | take 1"
+    response = KUSTO_CLIENT.execute(KUSTO_DATABASE, query)
+    data = dataframe_from_result_table(response.primary_results[0]).to_dict(
+        orient="records"
+    )
+    return data
+
+
+def get_host_assets():
+    query = """
+    EasmAsset
+    | where AssetName endswith '.gc.ca' or AssetName endswith '.canada.ca'
+    | where AssetType == 'HOST'
+    | summarize by AssetName, AssetUuid
+    """
+    response = KUSTO_CLIENT.execute(KUSTO_DATABASE, query)
+    data = dataframe_from_result_table(response.primary_results[0]).to_dict(
+        orient="records"
+    )
+    return data
+
+
+def get_host_assets_by_labels(label):
+    query = f"""
+    EasmAsset
+    | where AssetName endswith '.gc.ca' or AssetName endswith '.canada.ca'
+    | where AssetType == 'HOST'
+    | where Labels has '{label}'
+    | summarize by AssetName, AssetUuid, Labels
+    """
```

Comment:
Should look into using query parameters for KQL injection protection:
https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/query-parameters-statement?pivots=azuredataexplorer#example